src/osd/PrimaryLogPG.cc

```diff
@@ -3746,14 +3738,10 @@ int PrimaryLogPG::trim_object(
   } else {
     // save adjusted snaps for this object
     dout(10) << coid << " snaps " << old_snaps << " -> " << new_snaps << dendl;
-    if (legacy) {
-      coi.legacy_snaps = vector<snapid_t>(new_snaps.rbegin(), new_snaps.rend());
-    } else {
-      snapset.clone_snaps[coid.snap] = vector<snapid_t>(new_snaps.rbegin(),
-							new_snaps.rend());
-      // we still do a 'modify' event on this object just to trigger a
-      // snapmapper.update ... :(
-    }
+    snapset.clone_snaps[coid.snap] = vector<snapid_t>(new_snaps.rbegin(),
```

Comment:
use `emplace()` to save a ctor for temporary vector instance returned by `snapset.clone_snaps[coid.snap]`, like:
```c++
snapset.clone_snaps.emplace(
  coid.snap,
  vector<snapid_t>(new_snaps.rbegin(),new_snaps.rend()));
```

Reply:
y

Reply:
actually no, we are replacing an existing member, so emplace won't work.